### PR TITLE
Include locktoken in more logs

### LINKED ISSUE #22788: Update logging to include lockToken wherever available
It should be possible to trace a received message through the various touchpoints in the SDK. The best way to do this would be the lockToken as it will be specific to a particular instance of receiving the message (whereas SequenceNumber will always be the same for a given message each time it is received). Many events already include the lockToken, but we should log it whenever available.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/22788

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs [hunk 1]
```diff
@@ -809,30 +818,57 @@ public void ProcessorErrorHandlerThrewException(string exception)
...
             {
-                WriteEvent(ProcessorMessageHandlerCompleteEvent, identifier, sequenceNumber);
+                WriteEvent(ProcessorMessageHandlerStartEvent, identifier, sequenceNumber, lockToken);
             }
         }
 
-        [Event(ProcessorMessageHandlerExceptionEvent, Level = EventLevel.Error, Message = "{0}: User message handler complete: Message: SequenceNumber: {1}, Exception: {2}")]
-        public void ProcessorMessageHandlerException(string identifier, long sequenceNumber, string exception)
+        [NonEvent]
+        public void ProcessorMessageHandlerComplete(string identifier, long sequenceNumber, Guid lockToken)
```
Comment (pakrym) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs:840]: Does guid serialize ok?

Comment (pakrym) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs:840]: Should we just do string everywhere?

Comment (pakrym) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs:840]: My bad.

**Review by pakrym — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs:840]: This is a NonEvent.

**Comment by JoshLove-msft** [comment]
This is still missing from lock renewal events.

**Pull request merged**